tensorboard/plugins/pr_curve/pr_curves_plugin.py

```diff
@@ -0,0 +1,225 @@
+
+  def tags_impl(self):
+    """Creates the JSON object for the tags route response.
+
+    Returns:
+      The JSON object for the tags route response.
+    """
+    runs = self._multiplexer.Runs()
+    result = {run: {} for run in runs}
+
+    mapping = self._multiplexer.PluginRunToTagToContent(metadata.PLUGIN_NAME)
+    for (run, tag_to_content) in six.iteritems(mapping):
+      for (tag, content) in six.iteritems(tag_to_content):
+        summary_metadata = self._multiplexer.SummaryMetadata(run, tag)
+        result[run][tag] = {'displayName': summary_metadata.display_name,
+                            'description': plugin_util.markdown_to_safe_html(
+                                summary_metadata.summary_description)}
+
+    return result
+
+  @wrappers.Request.application
+  def available_time_entries_route(self, request):
+    """Gets a dict mapping run to a list of time entries.
+    Returns:
+      A dict with string keys (all runs with PR curve data). The values of the
+      dict are lists of time entries (consisting of the 3 fields below) to be
+      used in populating values within time sliders.
+    """
+    return http_util.Respond(
+        request, self.available_time_entries_impl(), 'application/json')
+
+  def available_time_entries_impl(self):
+    """Creates the JSON object for the available time entries route response.
+    Returns:
+      The JSON object for the available time entries route response.
+    """
+    all_runs = self._multiplexer.PluginRunToTagToContent(
+        metadata.plugin_name)
+    response = {}
+
+    # Compute the max step exhibited by any tag for each run. If a run lacks
+    # data, exclude it from the mapping.
+    for run, tag_to_content in all_runs.items():
+      if not tag_to_content:
+        # This run lacks data for this plugin.
+        continue
+
+      # Just use the list of tensor events for any of the tags to determine the
+      # steps to list for the run. The steps are often the same across tags for
+      # each run, albeit the user may elect to sample certain tags differently
+      # within the same run. If the latter occurs, TensorBoard will show the
+      # actual step of each tag atop the card for the tag.
+      tensor_events = self._multiplexer.Tensors(
+          run, min(six.iterkeys(tag_to_content)))
+      response[run] = [self._create_time_entry(e, tensor_events[0].wall_time)
+                       for e in tensor_events]
+    return response
+
+  def _create_time_entry(self, tensor_event, initial_wall_time):
+    """Creates a time entry given a tensor event.
+    
+    Arguments:
+      tensor_event: The tensor event for the time entry.
+      initial_wall_time: The wall time of the first event in seconds since the
+        epoch. Used to compute relative time.
+    """
+    return {
+      'step': tensor_event.step,
+      'wall_time': tensor_event.wall_time,
+      'relative': tensor_event.wall_time - initial_wall_time,
```

Comment:
Why compute the relative times on the server? No other plugin does this, as far as I know; we use `ChartHelpers.relativeAccessor` to do this on the frontend: https://github.com/tensorflow/tensorboard/blob/542ac06be716e8b28114bacfcb02535a608c1e12/tensorboard/plugins/scalar/vz_line_chart/vz-chart-helpers.ts#L155-L169

Reply:
That is true. The recent commit does away with computing relative on the backend. 

I think on the frontend, we could have pr-curve-card logic compute relative values. We could also generalize the logic within vz-chart-helpers, although the relative time in the PR curve UI changes the display in the sliders and doesn't modify the UI of the vz line chart.